docs/examples/openiotsdk_examples.md

```diff
@@ -583,86 +746,149 @@ Command Palette (F1), type: tasks <return>
 => Test Open IoT SDK example
 => Network namespace:  OIStestns
 => Network interface name: OIStesttap
 => Example application to use (lock-app)
 ```
 
-### Build and test the lock-app example using the CLI
-
-In CLI:
+### Build lock-app example with mbedtls crypto backend and lwip support in debug mode and debug it in the network namespace using the VSCode task
 
 ```
-${MATTER_ROOT}/scripts/examples/openiotsdk_example.sh -s -b psa -S lwip lock-app
+Command Palette (F1)
+=> Tasks: Run Task
+=> Build Open IoT SDK example
+=> Use debug mode (true)
+=> Use LwIP debug logs (false)
+=> Crypto algorithm to use (mbedtls)
+=> Socket API to use (lwip)
+=> Example application to use (lock-app)
 ```
 
 ```
-export TEST_NETWORK_NAME=OIStest
+Command Palette (F1)
+=> Tasks: Run Task
+=> Setup Open IoT SDK network
+=> Network namespace name: OIStest
+=> Command (restart)
 ```
 
 ```
-sudo ${MATTER_ROOT}/scripts/setup/openiotsdk/network_setup.sh -n $TEST_NETWORK_NAME restart
+1. Click 'Run and Debug' from the primary side menu or press (Ctrl+Shift+D)
+2. Select 'Debug Open IoT SDK example application' from the drop down list
+3. Click 'Start Debugging'(green triangle) or press (F5)
+4. Enter:
+=> lock-app
+=> GDB target address: 10.200.1.2
+=> Network namespace: OIStestns
+=> Network interface name: OIStesttap
+=> Example application to use (lock-app)
 ```
 
-```
-${MATTER_ROOT}/scripts/examples/scripts/run_in_ns.sh ${TEST_NETWORK_NAME}ns ${MATTER_ROOT}/scripts/examples/openiotsdk_example.sh -C test -n ${TEST_NETWORK_NAME}tap lock-app
-```
+Use debug controls to debug the application.
 
-### Build and debug the lock-app example using the VSCode task
+## Add new example
 
-```
-Command Palette (F1), type: tasks <return>
-=> Build Open IoT SDK example
-=> Use debug mode (True)
-=> Crypto algorithm to use (psa)
-=> Socket API to use (iotsocket)
-=> Example application to use (lock-app)
-```
+This chapter describes the details to add a new Matter example for the Open IoT
+SDK platform. Use the same `example_name` in each step below.
 
-In CLI:
+### Files structure
+
+A new example should be put into `examples/<example_name>/openiotsdk` directory.
+It should contain:
+
+-   application source files and headers in the `main` sub-directory
+-   application `CMakeLists.txt` file
+-   `.gitignore` file with with all sources to skip
+-   `README.md` file with example description
+-   additional directories with required configuration for used components. Use
+    `component_name-config` pattern, e.g `freertos-config`
+
+### Target name
+
+A new application target name should be created with
+`chip-openiotsdk-<example_name>-example(_ns)` pattern. The `_ns` suffix is
+required for [TF-M applications](#trusted-firmware-m).
+
+Example:
 
 ```
-export TEST_NETWORK_NAME=OIStest
+set(APP_TARGET chip-openiotsdk-new-example-example_ns)
 ```
 
+### Example tools
+
+Add a new example name to the
+[supported examples list](../../examples/platform/openiotsdk/supported_examples.md).
```

Comment:
I think we should mention that this file must be amended as well.

Reply:
What do you mean? Which file exactly?